apps/src/templates/EnhancedSafeMarkdown.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {connect} from 'react-redux';
+
+import SafeMarkdown from './SafeMarkdown';
+import {openDialog} from '@cdo/apps/redux/instructionsDialog';
+import {renderExpandableImages} from './utils/expandableImages';
+
+export class UnconnectedExpandableImagesWrapper extends React.Component {
+  static propTypes = {
+    showImageDialog: PropTypes.func.isRequired,
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node
+    ]).isRequired
+  };
+
+  componentDidMount() {
+    this.postRenderHook();
+  }
+
+  componentDidUpdate(prevProps) {
+    // TODO: do we need to do any kind of cleanup here? Or otherwise do
```

Comment:
I'm not sure if I hit this TODO I would know what to do with it

Reply:
I didn't either, that's why I left it as a todo rather than just doing the work myself. TODOs aren't intended to be calls to action, they're intended to be callouts to future engineers who find themselves in this part of the codebase. "If you're looking for something that could use some improvement, or hunting down a bug and are not sure where it might be, here's some code that could use attention"